### PR TITLE
nixos/iso-image: prepend to ISO menu labels

### PR DESCRIPTION
**Pull request by ners**
###### Description of changes

We can append to menu labels. Why not also prepend? :upside_down_face: 

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/installer/cd-dvd/iso-image.nix [hunk 1]
```diff
@@ -615,6 +615,18 @@ in
       '';
     };
 
+    isoImage.prependToMenuLabel = mkOption {
+      default = " Installer";
+      example = " Live System";
```
Comment (RaitoBezarius) [inline nixos/modules/installer/cd-dvd/iso-image.nix:620]: In that case, wouldn't the example say `Live System` ?

**Review by RaitoBezarius — COMMENTED** [review]
I'm not too shocked by this option, please follow CONTRIBUTING guide for your commit message and the title of the PR. But, can you provide a usecase for this option? Do you use it?

**Comment by ners** [comment]
Whoops, that was sloppy. Thanks for the catch, @RaitoBezarius and @pennae!

The use-case is that I would like to be able to specify either `<distroName> Installer` or `Install <distroName>`. And for NixOS it should indeed default to empty.

File: nixos/modules/installer/cd-dvd/iso-image.nix [hunk 2]
```diff
@@ -615,6 +615,18 @@ in
       '';
     };
 
+    isoImage.prependToMenuLabel = mkOption {
+      default = "";
+      example = "Install ";
```
Comment (bjornfor) [inline nixos/modules/installer/cd-dvd/iso-image.nix:620]: Please add `type = lib.types.str` in here.

**Comment by RaitoBezarius** [comment]
Result of `nixpkgs-review pr 223610` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
    <li>tests.nixos-functions.nixos-test</li>
  </ul>
</details>

**Review by RaitoBezarius — DISMISSED** [review]

**Comment by RaitoBezarius** [comment]
This PR does not eval anymore.
```
error: attribute 'type' already defined at /home/raito/.cache/nixpkgs-review/pr-223610-1/nixpkgs/nixos/modules/installer/cd-dvd/iso-image.nix:485:7

       at /home/raito/.cache/nixpkgs-review/pr-223610-1/nixpkgs/nixos/modules/installer/cd-dvd/iso-image.nix:490:7:

          489|       example = "zstd -Xcompression-level 6";
          490|       type = types.str;
             |       ^
          491|     };
(use '--show-trace' to show detailed location information)
```

**Comment by ners** [comment]
Fixed!

**Comment by RaitoBezarius** [comment]
Result of `nixpkgs-review pr 223610` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 tests built:</summary>
  <ul>
    <li>nixosTests.boot.biosUsb</li>
    <li>nixosTests.boot.uefiUsb</li>
  </ul>
</details>

**Comment by RaitoBezarius** [comment]
I'm experiencing Nix bugs that I am trying to fix while reviewing this PR, please ignore the nixpkgs-review in the meantime.

**Comment by RaitoBezarius** [comment]
Result of `nixpkgs-review pr 223610` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 tests built:</summary>
  <ul>
    <li>nixosTests.boot.biosUsb</li>
    <li>nixosTests.boot.uefiUsb</li>
  </ul>
</details>

**Comment by RaitoBezarius** [comment]
Good to go.

**Review by RaitoBezarius — APPROVED** [review]

**Pull request merged**